### PR TITLE
Enhancement/5524 finalize twg settings validation

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue:

- #5524

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Review by jimmymadon — CHANGES_REQUESTED** [review]
@eugene-manuilov The code and tests looks great! Just noticed that the SVGs are weird in shape (octagon) instead of circles. I think we need to export [the whole group](https://www.figma.com/file/t5zDpQ28Lo7ZyRcOpGdqgu/TwG?node-id=1781%3A12121) within each of the 'Color swatch circle' layers and not just the two rectangle elements. Thanks!

Screenshot 2022-07-24 at 16 15 30

**Comment by eugene-manuilov** [comment]
> The code and tests looks great! Just noticed that the SVGs are weird in shape (octagon) instead of circles. I think we need to export [the whole group](https://www.figma.com/file/t5zDpQ28Lo7ZyRcOpGdqgu/TwG?node-id=1781%3A12121) within each of the 'Color swatch circle' layers and not just the two rectangle elements. Thanks!

Thanks, @jimmymadon, but this is not necessary since these SVGs will be adjusted with CSS anyway. I have a PR #5606 that uses these images and I use CSS rules to add proper rounded borders: https://github.com/google/site-kit-wp/blob/3dd68dc91bf8b8c2911293108b3ff4c2cb0bde1b/assets/sass/components/thank-with-google/_googlesitekit-twg-color-radio.scss#L26

Adding additional ellipses to SVG files itself will only increase the size of those files.

**Review by jimmymadon — APPROVED** [review]
@eugene-manuilov Thanks for the clarification on the SVGs - makes sense. LGTM!

File: assets/js/modules/thank-with-google/components/settings/SettingsView.js [hunk 1]
```diff
@@ -29,18 +29,27 @@ import Data from 'googlesitekit-data';
...
+		select( MODULES_THANK_WITH_GOOGLE ).getColorTheme()
+	);
 
 	// Bail if the values aren't ready.
 	if ( publicationID === undefined ) {
 		return null;
 	}
 
+	const { name: colorName } =
+		getColorThemes()?.find(
```
Comment (techanvil) [inline assets/js/modules/thank-with-google/components/settings/SettingsView.js:49]: I don't see the optional chaining being necessary here, was this left in by accident?

```suggestion
		getColorThemes().find(
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Looks good @eugene-manuilov, just a very minor query from me, please take a look.

File: assets/js/modules/thank-with-google/components/settings/SettingsView.js (hunk 1 above)
Comment (eugene-manuilov) [reply assets/js/modules/thank-with-google/components/settings/SettingsView.js:49]: It was added "just in case". Should work without it 👍

**Review by techanvil — APPROVED** [review]
LGTM! Nice one @eugene-manuilov

**Pull request merged**